bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -235,6 +240,18 @@ protected void addDevice(Device device, String roomName) {
                 thingUID, thingTypeUID, device.id, device.deviceModel);
     }
 
+    /**
+     * Translates a Bosch device ID to an openHAB-compliant thing ID.
+     * <p>
+     * Characters that are not allowed in thing IDs are replaced by underscores.
+     * 
+     * @param deviceId the Bosch device ID
+     * @return the translated openHAB-compliant thing ID
+     */
+    private String buildCompliantThingID(String deviceId) {
+        return deviceId.replace(':', '_').replace('#', '_');
```

Comment:
This is not the same behaviour as before. Are you sure this won't change existing id's to be altered? That would be a breaking change.

Reply:
Yes, thanks for the clarification.